vllm/v1/executor/multiproc_executor.py

```diff
@@ -91,11 +93,21 @@ def sigusr1_handler(signum, frame):
         for w in self.workers:
             w.worker_response_mq.wait_until_ready()
 
+        # For pipeline parallel, we use a thread pool for asynchronous
+        # execute_model.
+        self.io_thread_pool: Optional[ThreadPoolExecutor] = None
+        if self.max_concurrent_batches > 1:
+            # Note: must use only 1 IO thread to keep dequeue sequence
+            # from the response queue
+            self.io_thread_pool = ThreadPoolExecutor(
+                max_workers=1, thread_name_prefix="mp_exec_io")
+
     def collective_rpc(self,
                        method: Union[str, Callable],
                        timeout: Optional[float] = None,
                        args: tuple = (),
-                       kwargs: Optional[dict] = None) -> list[Any]:
+                       kwargs: Optional[dict] = None,
+                       non_block: bool = False) -> list[Any]:
```

Comment:
since @comaniac and @ruisearch42  do not want to add `non_block` at the `collective_rpc` level, you can have a `_run_workers` function to have the ability to be `non_blocking`, and then in the `execute_model` level of this executor, call `_run_workers`with non-blocking if pp is enabled.

Reply:
Good idea :)